File: VB/GridDemo.Wpf/Controls/ColumnDescription.vb
Namespace GridDemo
    Public Class ColumnDescription
        Public Sub New(ByVal propertyName As String, Optional ByVal displayName As String = Nothing)
            Me.PropertyName = propertyName
            Me.DisplayName = If(displayName, propertyName)
        End Sub
        Private privatePropertyName As String
        Public Property PropertyName() As String
            Get
                Return privatePropertyName
            End Get
            Private Set(ByVal value As String)
                privatePropertyName = value
            End Set
        End Property
        Private privateDisplayName As String
        Public Property DisplayName() As String
            Get
                Return privateDisplayName
            End Get
            Private Set(ByVal value As String)
                privateDisplayName = value
            End Set
        End Property
    End Class
    Public Class BandDescription
        Public Sub New(ByVal displayName As String, ByVal columns() As ColumnDescription)
            Me.DisplayName = displayName
            Me.Columns = columns
        End Sub
        Private privateDisplayName As String
        Public Property DisplayName() As String
            Get
                Return privateDisplayName
            End Get
            Private Set(ByVal value As String)
                privateDisplayName = value
            End Set
        End Property
        Private privateColumns As ColumnDescription()
        Public Property Columns() As ColumnDescription()
            Get
                Return privateColumns
            End Get
            Private Set(ByVal value As ColumnDescription())
                privateColumns = value
            End Set
        End Property
    End Class
End Namespace

File: CS/GridDemo.Wpf/Controls/ColumnDescription.cs
namespace GridDemo {
    public class ColumnDescription {
        public ColumnDescription(string propertyName, string displayName = null) {
            PropertyName = propertyName;
            DisplayName = displayName ?? propertyName;
        }
        public string PropertyName { get; private set; }
        public string DisplayName { get; private set; }
    }
    public class BandDescription {
        public BandDescription(string displayName, ColumnDescription[] columns) {
            DisplayName = displayName;
            Columns = columns;
        }
        public string DisplayName { get; private set; }
        public ColumnDescription[] Columns { get; private set; }
    }
}
